tgui/packages/tgui/components/NumberInput.tsx

```diff
@@ -0,0 +1,351 @@
+    step: 1,
+    stepPixelSize: 1,
+    suppressFlicker: 50,
+  };
+
+  constructor(props: NumberInputProps) {
+    super(props);
+  }
+
+  suppressFlicker = () => {
+    const { suppressFlicker } = this.props;
+    if (suppressFlicker) {
+      this.setState({
+        suppressingFlicker: true,
+      });
+      clearTimeout(this.flickerTimer);
+      this.flickerTimer = setTimeout(
+        () =>
+          this.setState({
+            suppressingFlicker: false,
+          }),
+        suppressFlicker,
+      );
+    }
+  };
+
+  handleDragStart: MouseEventHandler<HTMLDivElement> = (event) => {
+    const { value, updateRate, disabled } = this.props;
+    const { editing } = this.state;
+    if (disabled || editing) {
+      return;
+    }
+    this.setState({
+      dragging: false,
+      origin: event.screenY,
+      internalValue: parseFloat(value.toString()),
+    });
+    this.dragTimeout = setTimeout(() => {
+      this.setState({
+        dragging: true,
+      });
+    }, 250);
+
+    this.dragInterval = setInterval(() => {
+      const { dragging, value } = this.state;
+      const { onDrag } = this.props;
+      if (dragging && onDrag) {
+        onDrag(event, value);
+      }
+    }, updateRate || DEFAULT_UPDATE_RATE);
+  };
+
+  handleDragMove: MouseEventHandler<HTMLDivElement> = (event) => {
+    const { minValue, maxValue, step, stepPixelSize, disabled } = this.props;
+    const { dragging } = this.state;
+    if (disabled || !dragging) {
+      return;
+    }
+
+    this.setState((prevState) => {
+      const state = { ...prevState };
+      const offset = state.origin - event.screenY;
+      if (prevState.dragging && step) {
+        const stepOffset = Number.isFinite(minValue) ? minValue % step : 0;
+        // Translate mouse movement to value
+        // Give it some headroom (by increasing clamp range by 1 step)
+        state.internalValue = clamp(
+          state.internalValue + (offset * step) / (stepPixelSize || 1),
+          minValue - step,
+          maxValue + step,
+        );
+        // Clamp the final value
+        state.value = clamp(
+          state.internalValue - (state.internalValue % step) + stepOffset,
+          minValue,
+          maxValue,
+        );
+        state.origin = event.screenY;
+      } else if (Math.abs(offset) > 4) {
+        state.dragging = true;
+      }
+      return state;
+    });
+  };
+
+  handleDragEnd: MouseEventHandler<HTMLInputElement> = (event) => {
+    const { value, dragging, internalValue } = this.state;
+    const { onDrag, onChange, disabled } = this.props;
+    if (disabled) {
+      return;
+    }
+
+    clearInterval(this.dragInterval);
+    clearTimeout(this.dragTimeout);
+    this.setState({
+      dragging: false,
+      editing: !dragging,
+    });
+
+    if (dragging) {
+      this.suppressFlicker();
+      if (onChange) {
+        onChange(event, value);
+      }
+      if (onDrag) {
+        onDrag(event, value);
+      }
```

Comment:
shave off some lines doing this here and elsewhere
```suggestion
      onChange?.(event, value);
      onDrag?.(event, value); 
```